src/main/kotlin/snyk/errorHandler/SnykErrorReportSubmitter.kt

```diff
@@ -0,0 +1,45 @@
+package snyk.errorHandler
+
+import com.intellij.diagnostic.AbstractMessage
+import com.intellij.diagnostic.PluginException
+import com.intellij.openapi.diagnostic.Attachment
+import com.intellij.openapi.diagnostic.ErrorReportSubmitter
+import com.intellij.openapi.diagnostic.IdeaLoggingEvent
+import com.intellij.openapi.diagnostic.SubmittedReportInfo
+import com.intellij.util.Consumer
+import java.awt.Component
+
+class SnykErrorReportSubmitter : ErrorReportSubmitter() {
+    override fun getReportActionText(): String {
+        return "Report to Snyk"
+    }
+
+    override fun submit(
+        events: Array<out IdeaLoggingEvent>,
+        additionalInfo: String?,
+        parentComponent: Component,
+        consumer: Consumer<SubmittedReportInfo>
+    ): Boolean {
+        for (event in events) {
+            var throwable = event.throwable
+            val description = additionalInfo ?: ""
+            var attachments = listOf<Attachment>()
+
+            if (event.data is AbstractMessage) {
+                throwable = (event.data as AbstractMessage).throwable
+                attachments = (event.data as AbstractMessage).includedAttachments
+            }
+            if (throwable is PluginException && throwable.cause != null) {
+                // unwrap PluginManagerCore.createPluginException
+                throwable = throwable.cause
+            }
+            SentryErrorReporter.submitErrorReport(
```

Comment:
This function always returns true. Should we return false when `SentryErrorReporter.submitErrorReport` fails? What's the behaviour when this.submit returns `false`?

Reply:
Just checked behavior when this.submit returns `false` and surprisingly nothing happened. I expect Idea at least notify user that submission was unsuccessful...
But anyway we probably should return from SentryErrorReporter.submitErrorReport if we actually been able to submit. And I see we already do log such event.

Reply:
Other plugins return 'true' always :), but let's not relay on others.

If submit returns `false` then your error for IntelliJ is still "active" and will/can be shown to users. I suggest that we show a notification whether submitting Sentry report was successful or not. Wdyt?

Reply:
I guess they return `true` for a better UX. If user gets a notification that it wasn't successful, what can he do with it? Probably nothing, additionally being annoyed that he cannot even submit an error report 😄 Maybe leave it as-is for the time being then :)